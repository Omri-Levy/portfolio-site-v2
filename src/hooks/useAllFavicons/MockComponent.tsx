import React from 'react';
import useAllFavicons from '~/hooks/useAllFavicons/useAllFavicons';

const MockComponent: React.FunctionComponent = () => {
	return <div>{JSON.stringify(useAllFavicons())}</div>;
};

export default MockComponent;
