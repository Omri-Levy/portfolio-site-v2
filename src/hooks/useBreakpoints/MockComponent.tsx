import React from 'react';
import useBreakpoints from '.';

const MockComponent: React.FunctionComponent = () => {
	const mediaQuery = useBreakpoints(`md`, `up`);

	return <div>{JSON.stringify(mediaQuery)}</div>;
};

export default MockComponent;
