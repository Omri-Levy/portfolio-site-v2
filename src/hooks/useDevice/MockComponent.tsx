import React from 'react';
import useDevice from '~/hooks/useDevice/useDevice';

const MockComponent: React.FunctionComponent = () => {
	const { isDesktop } = useDevice();

	return <div>{JSON.stringify(isDesktop)}</div>;
};

export default MockComponent;
