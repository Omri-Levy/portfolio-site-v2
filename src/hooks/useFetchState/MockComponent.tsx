import React from 'react';
import useFetchState from './useFetchState';

const MockComponent: React.FunctionComponent = () => {
	const {
		fetchState,
		setFetchState,
		isLoading,
		shouldDisplayAlert,
		severity,
		alertMessage,
	} = useFetchState({});

	return (
		<>
			{JSON.stringify(fetchState)}
			{JSON.stringify(setFetchState)}
			{JSON.stringify(isLoading)}
			{JSON.stringify(shouldDisplayAlert)}
			{JSON.stringify(severity)}
			{JSON.stringify(alertMessage)}
		</>
	);
};

export default MockComponent;
