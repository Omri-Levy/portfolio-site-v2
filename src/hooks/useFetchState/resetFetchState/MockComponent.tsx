import { Alert } from '@material-ui/lab';
import React, { useState } from 'react';
import resetFetchState from './resetFetchState';

const MockComponent: React.FunctionComponent = () => {
	const [fetchState, setFetchState] = useState<string | undefined>(`success`);

	if (fetchState === undefined) {
		return null;
	}

	return (
		<Alert onClose={resetFetchState(setFetchState, fetchState)}>
			test alert
		</Alert>
	);
};

export default MockComponent;
