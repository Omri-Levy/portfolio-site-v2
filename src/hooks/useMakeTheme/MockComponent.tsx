import React from 'react';
import useMakeTheme from '~/hooks/useMakeTheme/useMakeTheme';

const MockComponent: React.FunctionComponent = () => {
	const theme = useMakeTheme();

	return <div>{JSON.stringify(theme)}</div>;
};

export default MockComponent;
