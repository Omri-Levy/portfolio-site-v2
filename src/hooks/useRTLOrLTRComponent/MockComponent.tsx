import React, { useContext, useEffect, useState } from 'react';
import useRTLOrLTRComponent from './useRTLOrLTRComponent';
import { ThemeContext } from '~/context/ThemeContext';
import { Props } from './types';

const MockComponent: React.FunctionComponent<Props> = ({ rtl }) => {
	const { setIsRTL } = useContext(ThemeContext);
	const [isMounted, setIsMounted] = useState(true);

	useEffect(() => {
		if (rtl && isMounted) {
			setIsRTL(true);
		}

		return () => setIsMounted(false);
	}, []);

	return <div>{useRTLOrLTRComponent(<h1>rtl</h1>, <h1>ltr</h1>)}</div>;
};

export default MockComponent;
