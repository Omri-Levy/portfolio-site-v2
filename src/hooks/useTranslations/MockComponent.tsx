import React, { useEffect, useState } from 'react';
import useTranslations from '.';
import { Props } from './types';
import useThemeContext from '../../context/ThemeProvider/useThemeContext';

const MockComponent: React.FunctionComponent<Props> = ({ rtl }) => {
	const [isMounted, setIsMounted] = useState(true);
	const { setIsRTL } = useThemeContext();

	useEffect(() => {
		if (isMounted) {
			setIsRTL(rtl);
			if (typeof window !== `undefined`) {
				localStorage.setItem(`isRTL`, JSON.stringify(rtl));
			}
		}

		return () => setIsMounted(false);
	}, []);

	return <div>{useTranslations(`עברית`, `english`)}</div>;
};

export default MockComponent;
