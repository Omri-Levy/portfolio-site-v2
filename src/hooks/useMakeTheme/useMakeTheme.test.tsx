import { render } from '@testing-library/react';
import React from 'react';
import MockComponent from './MockComponent';

describe(`useMakeTheme`, () => {
	it(`returns a theme`, () => {
		const { container } = render(<MockComponent />);

		expect(container.firstChild).toMatchSnapshot();
	});
});
