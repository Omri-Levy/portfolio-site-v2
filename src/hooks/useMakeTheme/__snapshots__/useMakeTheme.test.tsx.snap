// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useMakeTheme returns a theme 1`] = `
<div>
  {"breakpoints":{"keys":["xs","sm","md","lg","xl"],"values":{"xs":350,"sm":600,"ms":800,"md":1000,"lg":1280,"mlg":1830,"xl":1920}},"direction":"ltr","mixins":{"toolbar":{"minHeight":56,"@media (min-width:350px) and (orientation: landscape)":{"minHeight":48},"@media (min-width:600px)":{"minHeight":64}}},"overrides":{"MuiCssBaseline":{"@global":{"body":{"overflowX":"hidden","direction":"ltr"},"img":{"width":"100%","display":"block"}}},"MuiSwitch":{"root":{"marginBottom":6},"thumb":{"color":"#3e96ff","marginTop":3,"width":20,"height":20},"track":{"backgroundColor":"#393e46","height":20,"width":40,"borderRadius":"100vh"},"colorSecondary":{"&$checked + $track":{"backgroundColor":"#eee"}}},"MuiButton":{"containedPrimary":{"color":"#fff","boxShadow":"0px 2px 4px #000","border":"1px solid rgb(46, 112, 191)","padding":4,"borderRadius":3}}},"palette":{"common":{"black":"#000","white":"#fff"},"type":"light","primary":{"main":"#3e96ff","border":"rgb(46, 112, 191)","secondary":"#222831","dark":"#393e46","contrastText":"#eee","light":"rgb(100, 171, 255)"},"secondary":{"main":"#3e96ff","light":"#fff","dark":"#000","contrastText":"#393e46"},"error":{"light":"#e57373","main":"#f44336","dark":"#d32f2f","contrastText":"#fff"},"warning":{"light":"#ffb74d","main":"#ff9800","dark":"#f57c00","contrastText":"rgba(0, 0, 0, 0.87)"},"info":{"light":"#64b5f6","main":"#2196f3","dark":"#1976d2","contrastText":"#fff"},"success":{"light":"#81c784","main":"#4caf50","dark":"#388e3c","contrastText":"rgba(0, 0, 0, 0.87)"},"grey":{"50":"#fafafa","100":"#f5f5f5","200":"#eeeeee","300":"#e0e0e0","400":"#bdbdbd","500":"#9e9e9e","600":"#757575","700":"#616161","800":"#424242","900":"#212121","A100":"#d5d5d5","A200":"#aaaaaa","A400":"#303030","A700":"#616161"},"contrastThreshold":3,"tonalOffset":0.2,"text":{"primary":"#fff","secondary":"#000","disabled":"rgba(0, 0, 0, 0.38)","hint":"rgba(0, 0, 0, 0.38)"},"divider":"rgba(0, 0, 0, 0.12)","background":{"paper":"#fff","default":"#222831"},"action":{"active":"rgba(0, 0, 0, 0.54)","hover":"rgba(0, 0, 0, 0.04)","hoverOpacity":0.04,"selected":"rgba(0, 0, 0, 0.08)","selectedOpacity":0.08,"disabled":"rgba(0, 0, 0, 0.26)","disabledBackground":"rgba(0, 0, 0, 0.12)","disabledOpacity":0.38,"focus":"rgba(0, 0, 0, 0.12)","focusOpacity":0.12,"activatedOpacity":0.12}},"props":{},"shadows":["none","0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12)","0px 3px 1px -2px rgba(0,0,0,0.2),0px 2px 2px 0px rgba(0,0,0,0.14),0px 1px 5px 0px rgba(0,0,0,0.12)","0px 3px 3px -2px rgba(0,0,0,0.2),0px 3px 4px 0px rgba(0,0,0,0.14),0px 1px 8px 0px rgba(0,0,0,0.12)","0px 2px 4px -1px rgba(0,0,0,0.2),0px 4px 5px 0px rgba(0,0,0,0.14),0px 1px 10px 0px rgba(0,0,0,0.12)","0px 3px 5px -1px rgba(0,0,0,0.2),0px 5px 8px 0px rgba(0,0,0,0.14),0px 1px 14px 0px rgba(0,0,0,0.12)","0px 3px 5px -1px rgba(0,0,0,0.2),0px 6px 10px 0px rgba(0,0,0,0.14),0px 1px 18px 0px rgba(0,0,0,0.12)","0px 4px 5px -2px rgba(0,0,0,0.2),0px 7px 10px 1px rgba(0,0,0,0.14),0px 2px 16px 1px rgba(0,0,0,0.12)","0px 5px 5px -3px rgba(0,0,0,0.2),0px 8px 10px 1px rgba(0,0,0,0.14),0px 3px 14px 2px rgba(0,0,0,0.12)","0px 5px 6px -3px rgba(0,0,0,0.2),0px 9px 12px 1px rgba(0,0,0,0.14),0px 3px 16px 2px rgba(0,0,0,0.12)","0px 6px 6px -3px rgba(0,0,0,0.2),0px 10px 14px 1px rgba(0,0,0,0.14),0px 4px 18px 3px rgba(0,0,0,0.12)","0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)","0px 7px 8px -4px rgba(0,0,0,0.2),0px 12px 17px 2px rgba(0,0,0,0.14),0px 5px 22px 4px rgba(0,0,0,0.12)","0px 7px 8px -4px rgba(0,0,0,0.2),0px 13px 19px 2px rgba(0,0,0,0.14),0px 5px 24px 4px rgba(0,0,0,0.12)","0px 7px 9px -4px rgba(0,0,0,0.2),0px 14px 21px 2px rgba(0,0,0,0.14),0px 5px 26px 4px rgba(0,0,0,0.12)","0px 8px 9px -5px rgba(0,0,0,0.2),0px 15px 22px 2px rgba(0,0,0,0.14),0px 6px 28px 5px rgba(0,0,0,0.12)","0px 8px 10px -5px rgba(0,0,0,0.2),0px 16px 24px 2px rgba(0,0,0,0.14),0px 6px 30px 5px rgba(0,0,0,0.12)","0px 8px 11px -5px rgba(0,0,0,0.2),0px 17px 26px 2px rgba(0,0,0,0.14),0px 6px 32px 5px rgba(0,0,0,0.12)","0px 9px 11px -5px rgba(0,0,0,0.2),0px 18px 28px 2px rgba(0,0,0,0.14),0px 7px 34px 6px rgba(0,0,0,0.12)","0px 9px 12px -6px rgba(0,0,0,0.2),0px 19px 29px 2px rgba(0,0,0,0.14),0px 7px 36px 6px rgba(0,0,0,0.12)","0px 10px 13px -6px rgba(0,0,0,0.2),0px 20px 31px 3px rgba(0,0,0,0.14),0px 8px 38px 7px rgba(0,0,0,0.12)","0px 10px 13px -6px rgba(0,0,0,0.2),0px 21px 33px 3px rgba(0,0,0,0.14),0px 8px 40px 7px rgba(0,0,0,0.12)","0px 10px 14px -6px rgba(0,0,0,0.2),0px 22px 35px 3px rgba(0,0,0,0.14),0px 8px 42px 7px rgba(0,0,0,0.12)","0px 11px 14px -7px rgba(0,0,0,0.2),0px 23px 36px 3px rgba(0,0,0,0.14),0px 9px 44px 8px rgba(0,0,0,0.12)","0px 11px 15px -7px rgba(0,0,0,0.2),0px 24px 38px 3px rgba(0,0,0,0.14),0px 9px 46px 8px rgba(0,0,0,0.12)"],"typography":{"htmlFontSize":16,"fontFamily":"Montserrat,sans-serif","fontSize":14,"fontWeightLight":300,"fontWeightRegular":400,"fontWeightMedium":500,"fontWeightBold":700,"h1":{"fontFamily":"Montserrat,sans-serif","fontWeight":300,"fontSize":"6rem","lineHeight":1.167,"color":"#fff"},"h2":{"fontFamily":"Montserrat,sans-serif","fontWeight":300,"fontSize":"3.75rem","lineHeight":1.2,"color":"#3e96ff"},"h3":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"3rem","lineHeight":1.167},"h4":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"2.125rem","lineHeight":1.235},"h5":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"1.5rem","lineHeight":1.334},"h6":{"fontFamily":"Montserrat,sans-serif","fontWeight":500,"fontSize":"1.25rem","lineHeight":1.6},"subtitle1":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"1rem","lineHeight":1.75},"subtitle2":{"fontFamily":"Montserrat,sans-serif","fontWeight":500,"fontSize":"0.875rem","lineHeight":1.57},"body1":{"fontFamily":"Open Sans,sans-serif","fontWeight":400,"fontSize":"1rem","lineHeight":1.5,"color":"#fff"},"body2":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"0.875rem","lineHeight":1.43},"button":{"fontFamily":"Montserrat,sans-serif","fontWeight":500,"fontSize":"0.875rem","lineHeight":1.75,"textTransform":"uppercase"},"caption":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"0.75rem","lineHeight":1.66},"overline":{"fontFamily":"Montserrat,sans-serif","fontWeight":400,"fontSize":"0.75rem","lineHeight":2.66,"textTransform":"uppercase"}},"shape":{"borderRadius":4},"transitions":{"easing":{"easeInOut":"cubic-bezier(0.4, 0, 0.2, 1)","easeOut":"cubic-bezier(0.0, 0, 0.2, 1)","easeIn":"cubic-bezier(0.4, 0, 1, 1)","sharp":"cubic-bezier(0.4, 0, 0.6, 1)"},"duration":{"shortest":150,"shorter":200,"short":250,"standard":300,"complex":375,"enteringScreen":225,"leavingScreen":195}},"zIndex":{"mobileStepper":1000,"speedDial":1050,"appBar":1100,"drawer":1200,"modal":1300,"snackbar":1400,"tooltip":1500}}
</div>
`;
